include/rocksdb/memory_allocator.h

```diff
@@ -36,9 +36,40 @@ class MemoryAllocator {
   }
 };
 
+namespace jemalloc {
+
+enum class PerCPUArena : int {
+  // Disable per-CPU arena. One single arena is used.
+  kDisabled = 0,
+
+  // Create one arena for each CPU, to reduce arena mutex contention,
+  // but may increase memory fragmentation.
+  kPerCPU = 1,
+
+  // Assume hyper-threading is enabled and create one arena for each physical
+  // CPU (half of #CPU in total). The allocator does not have run-time check to 
+  // verify whether the assumption is true. This mode may have less memory
+  // fragmentation comparing to kPerCPU, for using less arenas.
+  kPerPhysicalCPU = 2,
+};
+
+struct JemallocAllocatorOptions {
+  // If per-CPU arena is enabled, the allocator will create arena for each
+  // CPU. It reduces arena mutex contention and improves throughput, but
+  // potentially increases memory fragmentation.
+  PerCPUArena per_cpu_arena = PerCPUArena::kPerPhysicalCPU;
+
+  // Enable thread-specific cache (tcache) to reduce access to arenas,
+  // further reduce mutex contention with arena.
+  bool enable_tcache = true;
```

Comment:
Do we encourage anyone to set it to false? If not, we should remove this option.

Reply:
The cost of having the extra tcache should be relatively small (in terms of memory usage).  The edge case would be, if we have tons of threads accessing with low frequency, then it could consume a lot memory (may reach # of threads * ~1 MB) w/o bringing much benefit. If we don't foresee that as common / valid use cases, enabling by default should be fine.